include/boost/multiprecision/detail/endian.hpp

```diff
@@ -0,0 +1,45 @@
+////////////////////////////////////////////////////////////////
+//  Copyright 2021 Matt Borland. Distributed under the Boost
+//  Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
+
+#ifndef BOOST_MP_ENDIAN_HPP
+#define BOOST_MP_ENDIAN_HPP
+
+#include <boost/multiprecision/detail/standalone_config.hpp>
+
+#ifndef BOOST_MP_STANDALONE
+
+#  include <boost/predef/other/endian.h>
+#  define BOOST_MP_ENDIAN_BIG_BYTE BOOST_ENDIAN_BIG_BYTE
+#  define BOOST_MP_ENDIAN_LITTLE_BYTE BOOST_ENDIAN_LITTLE_BYTE
+
+#elif (__cplusplus >= 202002L || _MSVC_LANG >= 202002L)
+
+#  if __has_include(<bit>)
+#    include <bit>
+#    define BOOST_MP_ENDIAN_BIG_BYTE (std::endian::native == std::endian::big)
```

Comment:
Sadly, I don't believe this will work since the types and values used here are not usable in a preprocessor conditional?

Reply:
You are correct. `std::endian` is just an `enum` aliasing the byte order macros (with libstdc++ anyway).